README.md

```diff
@@ -131,113 +113,151 @@ This will output:
-   // running on localhost:8090
-   err = cm.MarkChannelAsGalactic(channel, "/topic/simple-stream", c)
-   if err != nil {
-      log.Panicf("unable to map local channel to broker destination: %e", err)
-   }
-
-   // wait for done signal
-   <-done
-
-   // mark channel as local (unsubscribe from all mappings)
-   err = cm.MarkChannelAsLocal(channel)
-   if err != nil {
-      log.Panicf("unable to unsubscribe, error: %e", err)
-   }
-   err = c.Disconnect()
-   if err != nil {
-      log.Panicf("unable to disconnect, error: %e", err)
-   }
+// SimpleStream will connect to our demo broker running at transport-bus.io, listen to a simple stream that is being
+// broadcast on /topic/simple-stream. Every second a random word is broadcast on that channel to anyone listening.
+// This should take 10 seconds to run.
+func SimpleStream() []string {
+
+	// get a pointer to the bus.
+	b := bus.GetBus()
+
+	// get a pointer to the channel manager
+	cm := b.GetChannelManager()
+
+	channel := "my-stream"
+	cm.CreateChannel(channel)
+
+	// listen to stream of messages coming in on channel, a handler is returned that allows you to add in
+	// lambdas that handle your
+	handler, err := b.ListenStream(channel)
+
+	if err != nil {
+		utils.Log.Panicf("unable to listen to channel stream, error: %v", err.Error())
+	}
+
+	// create a broker connector config and connect to transport-bus.io over WebSocket using TLS.
+	config := &bridge.BrokerConnectorConfig{
+		Username:   "guest",            // not required for demo, but our API requires it.
+		Password:   "guest",            // ^^ same.
+		ServerAddr: "transport-bus.io", // our live broker running plank and demo services.
+		UseWS:      true,               // connect over websockets
+		WebSocketConfig: &bridge.WebSocketConfig{ // configure websocket
+			WSPath: "/ws", // websocket endpoint
+			UseTLS: true,  // use TLS/HTTPS
+		}}
+
+	// connect to transport-bus.io demo broker
+	c, err := b.ConnectBroker(config)
+	if err != nil {
+		utils.Log.Fatalf("unable to connect to transport-bus.io, error: %v", err.Error())
+	}
+
+	// mark our local channel as galactic and map it to our connection and the /topic/simple-stream service
+	err = cm.MarkChannelAsGalactic(channel, "/topic/simple-stream", c)
+	if err != nil {
+		utils.Log.Fatalf("unable to map local channel to broker destination: %e", err)
+	}
+
+	// collect the streamed values in a slice
+	var streamedValues []string
+
+	// create a wait group that will wait 10 times before completing.
+	var wg sync.WaitGroup
+	wg.Add(10)
+
+	// keep listening
+	handler.Handle(
+		func(msg *model.Message) {
+
+			// unmarshal the message payload into a model.Response object
+			// this is a wrapper transport uses when being used as a server, it encapsulates a rich set of data
+			// about the message, but you only really care about the payload (body)
+			r := &model.Response{}
+			d := msg.Payload.([]byte)
```

Comment:
i think we can just go straight to `msg.Payload.(*model.Response)` but seems like you wanted more explanation here breaking it down to the smallest bits. is that the case?

Reply:
oh, it's because we're sending an encoded blob slice, yeah let's leave it like that.